utils_for_test.go

```diff
@@ -84,6 +85,71 @@ func writeFile(t *testing.T, prefix, contents string) string {
 	return f.Name()
 }
 
+type debugThrottler struct {
+	sync.Mutex
+	credits float64
+}
+
+func (d *debugThrottler) IsAllowed(operation string) bool {
+	const (
+		creditsPerDebugTrace = 1
+	)
+	d.Lock()
+	defer d.Unlock()
```

Comment:
nit: can you add a blank line after the `defer`

Reply:
Sure